server/src/main/java/org/elasticsearch/common/geo/builders/ShapeBuilder.java

```diff
@@ -211,7 +211,9 @@ protected JtsGeometry jtsGeometry(Geometry geom) {
      * the builder looses its validity. So this method should only be called once on a builder
      * @return new {@link Shape} defined by the builder
      */
-    public abstract T build();
+    public abstract T buildS4J();
+
+    public abstract Object buildLucene();
```

Comment:
Maybe in the future we want to add a new topology library, e.g. spatial3d, so I think it is better to name the method with a reference to the underlaying library.

Reply:
I think if we eventually end up supporting `spatial3d` or other packages then a more sophisticated factory pattern would be appropriate. But I chose this simple naming approach because its going to be removed / deleted in 7.0. So I didn't want to get too extravagant